calico_containers/tests/st/test_add_ip.py

```diff
@@ -13,54 +13,60 @@ def test_add_ip(self):
         """
         host = DockerHost('host')
 
-        host.execute("docker run -e CALICO_IP=192.168.1.1 -tid --name=node1 busybox", use_powerstrip=True)
-        host.execute("docker run -tid --name=node2 busybox")
-        host.calicoctl("container add node2 192.168.1.2 --interface=hello")
+        ip11 = "192.168.1.1"
+        ip12 = "192.168.1.2"
+        ip21 = "192.168.2.1"
+        ip22 = "192.168.2.2"
+        ip31 = "192.168.3.1"
+
+        node1 = host.create_workload("node1", ip11)
+        node2 = host.create_workload("node2")
+        host.calicoctl("container add %s %s --interface=hello" % (node2, ip12))
 
         host.calicoctl("profile add TEST_GROUP")
-        host.calicoctl("profile TEST_GROUP member add node1")
-        host.calicoctl("profile TEST_GROUP member add node2")
+        host.calicoctl("profile TEST_GROUP member add %s" % node1)
+        host.calicoctl("profile TEST_GROUP member add %s" % node2)
 
-        test_ping = partial(host.execute, "docker exec node1 ping 192.168.1.2 -c 1 -W 1")
+        test_ping = partial(node1.assert_can_ping, ip12)
```

Comment:
no need for partials here and elsewhere, since `assert_can_ping` has that function built-in.